script.js

```diff
@@ -1,12 +1,421 @@
 //You can edit ALL of the code here
-function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+
+function getData(URL) {
+  fetch(URL)
+    .then((response) => response.json())
+    .then((json) => {
+      makeShowPage(json);
+      selectShow(json);
+      search();
+      result();
+      home();
+      updateResult();
+      selectEpisode(json);
+    });
+}
+
+function setup(URL) {
```

Comment:
maybe `setup` is not the best name here, does it explain it clearly - something like `generatePageFromURL` may be more clear